lib/ansible/galaxy/collection/__init__.py

```diff
@@ -522,6 +534,14 @@ def download_collections(
     :param no_deps: Ignore any collection dependencies and only download the base requirements.
     :param allow_pre_release: Do not ignore pre-release versions when selecting the latest.
     """
+    for install_req in list(collections):
```

Comment:
This doesn't need to be converted to a list if it's already expected to be a list according to the typings.

Reply:
I'm mutating the list, don't want it changing while iterating, so I copied it.

Reply:
In that case, consider making the intent explicit:
```suggestion
    for install_req in collections.copy():
```

Reply:
Or
```suggestion
    for install_req in collections[:]:
```
_(whichever syntax you prefer)_

Reply:
This is because using `list()` suggests that the purpose is data conversion rather than copying and therefore it is confusing.

Reply:
:+1: Removed mutating the list while refactoring.